lib/cli/src/detect.test.js

```diff
@@ -1,27 +1,288 @@
+    },
+  },
+  {
+    name: projectTypes.MARIONETTE,
+    files: {
+      'package.json': {
+        dependencies: {
+          'backbone.marionette': '1.0.0',
+        },
+      },
+    },
+  },
+  {
+    name: projectTypes.MARKO,
+    files: {
+      'package.json': {
+        dependencies: {
+          marko: '1.0.0',
+        },
+      },
+    },
+  },
+  {
+    name: projectTypes.RIOT,
+    files: {
+      'package.json': {
+        dependencies: {
+          riot: '1.0.0',
+        },
+      },
+    },
+  },
+  {
+    name: projectTypes.PREACT,
+    files: {
+      'package.json': {
+        dependencies: {
+          preact: '1.0.0',
+        },
+      },
+    },
+  },
+  {
+    name: projectTypes.RAX,
+    files: {
+      '.rax': 'file content',
+      'package.json': {
+        dependencies: {
+          rax: '1.0.0',
+        },
+      },
+    },
+  },
+];
+
+describe('Detect', () => {
+  it(`should return type ${projectTypes.HTML} if html option is passed`, () => {
+    getPackageJson.mockImplementation(() => true);
+    expect(detect({ html: true })).toBe(projectTypes.HTML);
+  });
+
+  it(`should return type ${projectTypes.UNDETECTED} if neither packageJson or bowerJson exist`, () => {
+    getPackageJson.mockImplementation(() => false);
+    getBowerJson.mockImplementation(() => false);
+    expect(detect()).toBe(projectTypes.UNDETECTED);
   });
-  it('false if no devDependency', () => {
-    expect(isStorybookInstalled({}, false)).toBe(false);
+
+  it(`should return language ${supportedLanguages.TYPESCRIPT} if the dependency is present`, () => {
+    getPackageJson.mockImplementation(() => ({
+      dependencies: {
+        typescript: '1.0.0',
+      },
+    }));
+    expect(detectLanguage()).toBe(supportedLanguages.TYPESCRIPT);
+  });
+
+  it(`should return language ${supportedLanguages.JAVASCRIPT} by default`, () => {
+    getPackageJson.mockImplementation(() => true);
+    expect(detectLanguage()).toBe(supportedLanguages.JAVASCRIPT);
   });
 
-  supportedFrameworks.forEach((framework) => {
-    it(`true if devDependencies has ${framework} Storybook version`, () => {
-      const devDependencies = {};
-      devDependencies[`@storybook/${framework}`] = '4.0.0-alpha.21';
-      expect(isStorybookInstalled({ devDependencies }, false)).toBeTruthy();
+  describe('isStorybookInstalled should return', () => {
+    it('false if empty devDependency', () => {
+      expect(isStorybookInstalled({ devDependencies: {} }, false)).toBe(false);
+    });
+
+    it('false if no devDependency', () => {
+      expect(isStorybookInstalled({}, false)).toBe(false);
+    });
+
+    supportedFrameworks.forEach((framework) => {
+      it(`true if devDependencies has ${framework} Storybook version`, () => {
+        const devDependencies = {
+          [`@storybook/${framework}`]: '4.0.0-alpha.21',
+        };
+        expect(isStorybookInstalled({ devDependencies }, false)).toBeTruthy();
+      });
+    });
+
+    it('true if forced flag', () => {
```

Comment:
this doesn't have the forced flag and doesn't return true?

Reply:
Great catch! That was one of the existing tests that I just moved inside a describe. Checking it better, it made no sense, as the method either returns false or a project type. Changed + fixed the other tests which were wrong and added one more!